qiita_pet/templates/study_base.html

```diff
@@ -15,12 +16,26 @@
 
   function fill_artifact(aid) {
     $("#study-main").html("ARTIFACT " + aid + " HTML STUFF");
+    $.get("/admin/artifact/", {artifact_id: +aid })
```

Comment:
everything is still passed as a string to the server, so yes, both checks are needed. Client side to make sure an integer is passed, and server side to make sure it is cast to integer after it is received.